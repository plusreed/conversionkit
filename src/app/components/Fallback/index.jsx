import React from 'react'
import Spinner from 'react-bootstrap/Spinner'

const Fallback = props => (
    <div className="p-2 mx-auto text-center bg-gray-900 w-auto">
        <h1 className="font-bold text-xl text-gray-100">
            {
                props.text ||
                <Spinner animation="border" role="status">
                    <span className="sr-only">Loading...</span>
                </Spinner>
            }
        </h1>
    </div>
)

export default Fallback
