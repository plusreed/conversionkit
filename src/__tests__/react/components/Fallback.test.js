import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import Fallback from '../../../app/components/Fallback';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Fallback />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders text correctly', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Fallback text={"test"} />, div);
  expect(div.innerHTML).toMatch(/test/)
})

it('displays a spinner when no text is provided', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Fallback />, div)
  // when the spinner is active, a screen reader element is added.
  // this is the best way i know to check if it's working properly.
  expect(div.innerHTML).toMatch(/sr-only/)
})

it('matches snapshot when no text is provided', () => {
  const tree = renderer
    .create(<Fallback />)
    .toJSON()

  expect(tree).toMatchSnapshot()
});

it('matches snapshot when text is provided', () => {
  const tree = renderer
    .create(<Fallback text={"test"} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
});