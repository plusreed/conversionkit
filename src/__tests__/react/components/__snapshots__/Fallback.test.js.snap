// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when no text is provided 1`] = `
<div
  className="p-2 mx-auto text-center bg-gray-900 w-auto"
>
  <h1
    className="font-bold text-xl text-gray-100"
  >
    <div
      className="spinner-border"
      role="status"
    >
      <span
        className="sr-only"
      >
        Loading...
      </span>
    </div>
  </h1>
</div>
`;

exports[`matches snapshot when text is provided 1`] = `
<div
  className="p-2 mx-auto text-center bg-gray-900 w-auto"
>
  <h1
    className="font-bold text-xl text-gray-100"
  >
    test
  </h1>
</div>
`;
